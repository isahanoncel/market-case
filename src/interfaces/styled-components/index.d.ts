// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft' | 'Check';

export type Colors =
  | 'primary'
  | 'gray'
  | 'secondary'
  | 'whiteBlue'
  | 'white'
  | 'background'
  | 'transparent'
  | 'innerBorder'
  | 'black100'
  | 'thinGray'
  | 'black500'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
