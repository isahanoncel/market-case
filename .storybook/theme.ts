import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
