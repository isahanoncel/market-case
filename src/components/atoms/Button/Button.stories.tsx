import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Button from '.';

export default {
  title: 'Atoms/Button',
  component: Button
} as ComponentMeta<typeof Button>;

export const Default: ComponentStory<typeof Button> = (args) => (
  <Button {...args}>Button</Button>
);
