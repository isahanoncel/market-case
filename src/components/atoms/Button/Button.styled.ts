import styled from 'styled-components';

interface IStyledButton {
  backgroundColor?: string;
  color?: string;
  margin?: string;
  fullWidth?: boolean;
}

export const StyledButton = styled.button<IStyledButton>`
  ${(props) => props.disabled && 'opacity:0.7;'};
  ${(props) => props.margin && `margin:${props.margin};`}
  ${(props) => props.fullWidth && `width:100%;`}

  ${(props) =>
    props.backgroundColor && `background-color: ${props.backgroundColor} ;`}
  ${(props) => props.color && `color: ${props.color} ;`}
  border: none;
  font-weight: 600;
  padding: 6px 16px;
  border-radius: 2px;
  cursor: pointer;
  min-width: 85px;
  min-height: 32px;
`;
