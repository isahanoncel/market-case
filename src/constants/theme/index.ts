import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
