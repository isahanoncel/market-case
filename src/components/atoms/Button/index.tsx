import React, { FC, useCallback } from 'react';
import { COLORS } from '../../../constants/theme';
import * as S from './Button.styled';

interface IButton {
  onClick?: () => void;
  children: string;
  disabled?: boolean;
  margin?: string;
  secondary?: boolean;
  fullWidth?: boolean;
}
const Button: FC<IButton> = ({
  children,
  onClick,
  disabled,
  margin,
  secondary,
  fullWidth,
}) => {
  const buttonTheme = useCallback(() => {
    if (secondary) {
      return {
        color: COLORS.primary,
        backgroundColor: COLORS.whiteBlue,
      };
    }
    return {
      color: COLORS.white,
      backgroundColor: COLORS.primary,
    };
  }, [secondary]);

  return (
    <S.StyledButton
      disabled={disabled}
      fullWidth={fullWidth}
      backgroundColor={buttonTheme().backgroundColor}
      color={buttonTheme().color}
      onClick={onClick}
      margin={margin}
    >
      {children}
    </S.StyledButton>
  );
};

export default Button;
