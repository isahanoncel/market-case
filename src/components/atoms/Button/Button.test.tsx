import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Button from '.';
import theme from '../../../constants/theme';

test('Button snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button>Button</Button>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
