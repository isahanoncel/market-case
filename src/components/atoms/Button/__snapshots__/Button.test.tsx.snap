// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing 1`] = `
.c0 {
  background-color: #1EA4CE;
  color: #fff;
  border: none;
  font-weight: 600;
  padding: 6px 16px;
  border-radius: 2px;
  cursor: pointer;
  min-width: 85px;
  min-height: 32px;
}

<button
  className="c0"
  color="#fff"
>
  Button
</button>
`;
